### PR TITLE
Add FXIOS-10462  Enhanced Tracking Protection Connection Status Refresh

### PR DESCRIPTION
**Pull request by petruSt24**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-10462)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/22896)

## :bulb: Description

Remade the previous [PR](https://github.com/mozilla-mobile/firefox-ios/pull/22899)  after a bad rebase, added only the refresh action in this one but I have another PR that handles the back/dismiss navigation in the BrowserViewController
## :pencil: Checklist
You have to check all boxes before merging
- [X] Filled in the above information (tickets numbers and description of your work)
- [X] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [X] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

**Comment by cyndichin** [comment]
thanks @petruSt24! I believe this PR also has the navigation logic added as well, not sure if that was intentional or if it will be updated to only include refresh logic. et me know if this PR is still ready for review. If so, my comment from the previous PR applies here too. https://github.com/mozilla-mobile/firefox-ios/pull/22899

**Comment by petruSt24** [comment]
@cyndichin yes this will be updated to have the navigation based on the example you provided, but will do it in another PR as I'll still need the new actions from this one

File: firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift
Comment (cyndichin) [inline firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift:202]: since this will be addressed in a separate PR, should we put a TODO instead here instead of implementing the navigation logic?

Comment (petruSt24) [reply firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift:202]: I would add a TODO but removing the navigation code would render the navigation bar unresponsive until the other PR is merged, is that preferred?

Comment (cyndichin) [reply firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift:202]: I think that's okay and we can add a `TODO` since this feature is still a work in progress + so we merge code that we're confident that we want to keep in.

**Comment by cyndichin** [comment]
@petruSt24 please re-request me to review when this is ready for another look! I will be on PTO starting 11/21, so I will defer to @nbhasin2 to continue to review if we don't get it merge before then. Thanks!

**Review by MattLichtenstein — CHANGES_REQUESTED** [review]
Bitrise has a couple of failures:

`TrackingProtectionViewController.swift:628:37: Value of type 'TrackingProtectionConnectionDetailsView' has no member 'applyTheme'`

`TrackingProtectionViewController.swift:139:106: Cannot convert return expression of type 'TrackingProtectionMiddlewareAction' to return type 'TrackingProtectionAction'`

File: firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift
Comment (MattLichtenstein) [inline firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift:195]: I believe we will need to remove this since it will result in a compiler warning and we have warning set to be treated as errors.

**Review by MattLichtenstein — CHANGES_REQUESTED** [review]

**Review by MattLichtenstein — APPROVED** [review]

**Comment by nbhasin2** [comment]
@petruSt24 this is approved, and you fixed test are you waiting for BR?

**Comment by petruSt24** [comment]
@nbhasin2 yes, it needs another bitrise run

**Pull request merged**
